contrib_bots/lib/DefineBot/DefineBot.py

```diff
@@ -0,0 +1,71 @@
+# See readme.md for instructions on running this code.
+import json
+import requests
+
+class DefineHandler(object):
+    '''
+    This plugin facilitates creating define tasks when
+    you are using Zulip to conduct a virtual meeting.  It
+    looks for messages starting with '@define'.
+
+    In this example, we write define items to a special
+    Zulip stream called "define", but this code could
+    be adapted to write define items to some kind of
+    external issue tracker as well.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to flag messages
+            as being define items.  Users should preface
+            messages with "@define".
+
+            Before running this, make sure to create a stream
+            called "define" that your API user can send to.
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        
+        is_define = original_content.startswith('@define')
+
+        return is_define
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        # Remove '@define' from the message and extract the rest of the message, the
+        # word/phrase to define.
+        splitContent = original_content.split(' ')
+        toDefine = ' '.join(splitContent[1:])
+
+        # Use OwlBot API to fetch definition.
+        api_result = requests.get("https://owlbot.info/api/v1/dictionary/%s?format=json" % (toDefine))
+        # Convert API result from string to JSON format.
+        definitions = json.loads(api_result.text)
+
+        response = '**%s**:\n' % (toDefine)
+
+        # Could not fetch definitions for the given word/phrase.
+        if definitions == []:
+            response += "Definition not available."
+        else: # Definitions available.
+            # Show definitions line by line.
+            for definition in definitions:
+                response += "\n(%s) %s" % (definition['type'], definition['defenition'])
```

Comment:
I think there is a typo in `definition['defenition']`. Using `+=` on strings in Python in a loop is inefficient. You should concatenate like this: 
```Python
response += '\n'.join("({}) {}".format(d['type'], d['definition']) for d in definitions)
```